bgp-monitor-action-pack/bgp-monitor/script.py

```diff
@@ -55,6 +56,31 @@ def IsStatsDiffExpected(prevStats, currentStats, expectedDiff: int):
         err = "Missing change control device" if device is None \
             else f"device {device} is missing 'id'"
         raise ActionFailed(err)
+
+    # This block checks for the check_established flag, defaults to False
+    # If set to True, this block will check that all BGP peers are "Established"
+    # If any BGP peers are not Established, the CC will abort/fail
+    if checkEstablished:
+        bgpPeerState = ctx.runDeviceCmds(['enable', 'show ip bgp summary vrf all', 'show bgp evpn summary'])
+        errs = [resp.get('error') for resp in bgpPeerState if resp.get('error')]
+        if errs:
+            raise ActionFailed(f"Running action failed with: {errs[0]}")
+
+        for bgppeer in bgpPeerState[1]["response"]["vrfs"]["default"]["peers"]:
+            if  bgpPeerState[1]["response"]["vrfs"]["default"]["peers"][bgppeer]["peerState"] != "Established":
+                establishedState = False
```

Comment:
This boolean is unneeded as we're causing a failure immediately afterwards, and we never really need to leverage it

Reply:
Yes, that is something I was seeing as I got it working. I will re-work this.